salt/elasticsearch/init.sls

```diff
@@ -36,3 +36,34 @@ elasticsearch-elasticsearch-yml:
         - name: /etc/elasticsearch/elasticsearch.yml
         - source: salt://elasticsearch/elasticsearch.yml
         - template: jinja
+
+
+{% set user = '_apt' if grains['os_family'] == 'Debian' and grains['osmajorrelease']|int >= 9 else 'root' %}
+
+{% for family in ('ipv4', 'ipv6') %}
+elasticsearch-outbound-firewall-{{ family }}:
+    firewall.append:
+        - table: filter
+        - chain: OUTPUT
+        - family: {{ family }}
+        - proto: tcp
+        - dport: 9300
+        - match:
+            - comment
+            - owner
+        - comment: 'elasticsearch: Allow outgoing traffic for for internal comms'
+        - uid-owner: {{ user }}
```

Comment:
Harcode `elasticsearch` here and add a
```yaml
- require:
    - pkg: elasticsearch
```
to make sure the user exists before applying the rules. You should also include the DNS permissions as I'm planning to use that from the start.